ocaml/idl/datamodel.ml

```diff
@@ -5249,14 +5249,46 @@ let network_detach_for_vm = call
     ~allowed_roles:_R_VM_POWER_ADMIN
     ()
 
+let network_purpose = Enum ("network_purpose", [
+  "nbd", "Network Block Device service using TLS";
+  "insecure_nbd", "Network Block Device service without integrity or confidentiality: NOT RECOMMENDED";
+  "management", "Management";
+  "storage", "Storage";
+  "guest", "Guest virtual machines";
+  "himn", "Host internal management network on which guests will be assigned a private link-local IP address which can be used to talk XenAPI";
+  "unmanaged", "Not managed by xapi";
+])
+(*
```

Comment:
Why commented out? I think it is clearer to add these functions in the patch that actually introduces them into the datamodel.